### PR TITLE
Update/better env variable handling

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at handling the env variables better with parsing and default values being centralized in one place.

### CODE REVIEW

File: packages/cw-orch-core/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 pub mod contract;
 pub mod env;
+pub use env as CwOrchEnvVars;
```
Comment (CyberHoward) [inline packages/cw-orch-core/src/lib.rs:3]: oeeehh yikes! Have you seen this pattern/naming somewhere before? I don't really like it. People will think it's a type when it isn't.

File: packages/cw-orch-core/src/env.rs
Comment (CyberHoward) [inline packages/cw-orch-core/src/env.rs:1]: I was thinking you were going to add values to the env enums. 

I.e. 

```rust
enum EnvVar {
    Abc(String),
    Xyz(u8),
}
```

I think this crate might be of interest: https://github.com/Peternator7/strum

If you could give it a look and still think this is good then it's also good for me!

Comment (CyberHoward) [reply packages/cw-orch-core/src/env.rs:1]: This one specifically looks interesting: https://docs.rs/strum_macros/0.25.3/strum_macros/derive.EnumDiscriminants.html

File: packages/cw-orch-core/src/lib.rs (hunk 1 above)
Comment (Kayanski) [reply packages/cw-orch-core/src/lib.rs:3]: Yeah you're right, will change

File: packages/cw-orch-core/src/env.rs
Comment (Kayanski) [reply packages/cw-orch-core/src/env.rs:1]: One issue why I chose this implementation is that I can't get a typed value from an enum variant.

For instance in your example, I can't have a at the same time : 
- EnvVar::Abc.value() method that gets me a String and 
- EnvVar::Xyz.value() method that gets me a u8

Comment (CyberHoward) [reply packages/cw-orch-core/src/env.rs:1]: The idea would be that you can't construct the type without it having its value. 

```rust
EnvVar {
   A(String),
   B(u16),
   ...
```

And then use a crate like this: https://docs.rs/strum_macros/0.25.3/strum_macros/derive.EnumTryAs.html
to get its value when you need it.

**Pull request merged**